### PR TITLE
Bug fix - Lab & Luv color conversion nans in gradient

### PR DESCRIPTION
**Pull request by cceyda**
### Description
Bug fix for https://github.com/kornia/kornia/issues/868
- `torch.pow(x=0)` ops were causing nans. solved by using `x.clamp_min(threshold)` (doesn't effect result values)
- Moved the Linear RGB<->sRGB conversion used in color conversions to rgb module. They are commonly used in luv and lab conversions. Previously the terms sRGB <-> linear rgb were mistakenly referred to in reverse, now fixed [Ref](https://stackoverflow.com/questions/35952564/convert-rgb-to-srgb)

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [x] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [x] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [x] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [x] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/color/lab.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import torch.nn as nn
 
 from .xyz import rgb_to_xyz, xyz_to_rgb
+from .rgb import rgb_to_linear_rgb, linear_rgb_to_rgb
```
Comment (cceyda) [inline kornia/color/lab.py:7]: I'm not really liking these function names `rgb_to_linear_rgb, linear_rgb_to_rgb` although `srgb_to_linear_rgb, linear_rgb_to_srgb` would have been more accurate. I stuck with just RGB for consistency. maybe we should mention somewhere in the docs that kornia assumes sRGB as the default RGB format.

Comment (edgarriba) [reply kornia/color/lab.py:7]: and what about simply `srgb_to_rgb` <-> `rgb_to_srgb` ?

File: kornia/color/lab.py [hunk 2]
```diff
@@ -37,27 +38,20 @@ def rgb_to_lab(image: torch.Tensor) -> torch.Tensor:
...
+    xyz_im: torch.Tensor = rgb_to_xyz(lin_rgb)
 
     # normalize for D65 white point
     xyz_ref_white = torch.tensor(
         [0.95047, 1., 1.08883], device=xyz_im.device, dtype=xyz_im.dtype)[..., :, None, None]
     xyz_normalized = torch.div(xyz_im, xyz_ref_white)
 
-    power = torch.pow(xyz_normalized, 1 / 3)
+    threshold = 0.008856
+    power = torch.pow(xyz_normalized.clamp_min(threshold), 1 / 3.)
```
Comment (edgarriba) [inline kornia/color/lab.py:52]: not whether `clamp_min` will be supported in older versions. I guess we could alternatively use `.clamp(min=threshold)`

File: kornia/color/rgb.py [hunk 3]
```diff
@@ -169,6 +169,55 @@ def rgba_to_bgr(image: torch.Tensor) -> torch.Tensor:
     return rgb_to_bgr(x_rgb)
 
 
+def rgb_to_linear_rgb(image: torch.Tensor) -> torch.Tensor:
+    r"""Convert an sRGB image to linear RGB. Used in colorspace conversions.
```
Comment (edgarriba) [inline kornia/color/rgb.py:173]: ok, given this function definition we should definitely name this function as `srgb_to_rgb`

File: kornia/color/rgb.py [hunk 4]
```diff
@@ -169,6 +169,55 @@ def rgba_to_bgr(image: torch.Tensor) -> torch.Tensor:
...
+    Args:
+        image (torch.Tensor): sRGB Image to be converted to linear RGB of shape :math:`(*,3,H,W)`.
+
+    Returns:
+        torch.Tensor: linear RGB version of the image with shape of :math:`(*,3,H,W)`.
+
+    Example:
+        >>> input = torch.rand(2, 3, 4, 5)
+        >>> output = rgb_to_linear_rgb(input) # 2x3x4x5
+    """
```
Comment (edgarriba) [inline kornia/color/rgb.py:184]: could we add basic type and shape check here ?

**Review by edgarriba — COMMENTED** [review]
This is great! Minor comments and we need test for the new functions and possibly a specific test that shows that the raised issue is solved. Let me know if you need any help - we are releasing this week and this fix should be definitely in.

File: kornia/color/lab.py (hunk 1 above)
Comment (cceyda) [reply kornia/color/lab.py:7]: `srgb_to_rgb` might be more confusing because `linear_rgb != rgb`. 
I'm leaving it as it is because we use `srgb==rgb` anyway

File: kornia/color/lab.py (hunk 2 above)
Comment (cceyda) [reply kornia/color/lab.py]: I'm a slave to copy/paste 🤣 replaced all with clip

**Comment by cceyda** [comment]
- added tests for `rgb_to_linear_rgb` &` linear_rgb_to_rgb`. It looked like some stuff for `bgr_to_rgb` were missing in tests so added those I saw was missing. On second thought I decided to leave the function names as is, better be explicit than vague, because I don't want there to be confusion between `sRGB`, `RGB` and `linear RGB`. In kornia when we use/say RGB we mean sRGB and `linear RGB` is the non-gamma-rectified version of sRGB. [ref](https://stackoverflow.com/questions/35952564/convert-rgb-to-srgb)
 -  modified test_lab and test_luv to include edge values [0,1] in the unit tests, since that's where it failed before. I had thought `test_gradcheck` also caught nans in the gradient but doesn't seem so 🤔 Maybe we should do some of the checks on real images vs randomly generated values, but then test costs might be higher.  

LGTM for now

**Review by edgarriba — APPROVED** [review]
LGTM
